src/lib/y2users/user.rb

```diff
@@ -89,32 +122,29 @@ def expire_date
       password&.account_expiration
     end
 
-    # @return [String] Returns full name from gecos entry or username if not specified in gecos.
+    # User full name
+    #
+    # It is extracted from GECOS if possible. Otherwise, the user name is considered as the full
+    # name.
+    #
+    # @return [String]
     def full_name
       gecos.first || name
     end
 
-    ATTRS = [:name, :uid, :gid, :shell, :home].freeze
-
-    # Clones user to different configuration object.
-    # @return [Y2Users::User] newly cloned user object
-    def clone_to(config)
-      attrs = ATTRS.each_with_object({}) { |a, r| r[a] = public_send(a) }
-      attrs.delete(:name) # name is separate argument
-      cloned = self.class.new(config, name, attrs)
-      cloned.password = password.clone_to(config) if password
-
-      cloned
-    end
-
-    # Compares user object if all attributes are same excluding configuration reference.
-    # @return [Boolean] true if it is equal
+    # Whether two users are equal
+    #
+    # Only relevant attributes are compared. For example, the config in which the user is attached
+    # and the internal user id are not considered.
+    #
+    # @return [Boolean]
```

Comment:
Note: so, this method cannot be used to infer _new_ users given two configurations, the "origin" and the "cloned". Instead, the `#is?` method from `ConfigElement` should be used. Right?

Reply:
The only way to be sure that two users are actually the same is by means of the id. We cannot rely on the rest of attributes. So, `#is?` should be used to find new users, and `#==` to find changes between two users. Anyway, facility methods can be added to the API in order to compare configs.